cmd/infra/powervs/create.go

```diff
@@ -1159,13 +1165,35 @@ func (infra *Infra) setupPowerVSDHCP(ctx context.Context, options *CreateInfraOp
 	return nil
 }
 
+// isRetryableError validates err contains possible retryable error keywords like timeoutErrorKeywords, if yes, the action can be retried and increase the retryCount, if not return the error.
+// if retryCount exceeds retryLimit, a retry limit exceeded error will be thrown.
+func isRetryableError(err error, retryableErrorKeywords []string, retryCount *int) error {
+	var retry bool
+	for _, e := range retryableErrorKeywords {
+		if strings.Contains(err.Error(), e) {
+			retry = true
+			break
+		}
+	}
+	if retry {
+		if *retryCount > retryLimit {
+			return fmt.Errorf("retry limit exceeded on error: %w", err)
+		}
+		*retryCount += 1
+		return nil
+	}
```

Comment:
I feel pooling method does a same task, can we have ignore error only part of this flow and let polling method to let polling method itself to deal with this based on the timeout.

Reply:
I thought of limiting the retryable errors too with some limit. If we skip in case of retryable error and depend on the timeout of polling method, it may retry on the whole duration until the timeout occurs. In case of DHCP monitoring we have timeout of 30mins.

Reply:
if it is not that flaky then we are good, if its fails forever then we have an issues. Have you come across a scenario it failed always with 5xx error in that window.?

Reply:
Not really, I have n't seen its consistently repeating.

Reply:
Ok, then maybe will remove the retryLimit logic and in future if we want to limit for some reason based on some scenario, we can maintain this count on calling func itself I guess.

Reply:
refactored the code, ptal!